src/programs/programs.service.ts

```diff
@@ -34,16 +35,44 @@ export class ProgramsService {
     }
   }
 
-  findAll() {
-    return this.dbService.program.findMany();
+  async findAll() {
+    let programs = await this.dbService.program.findMany();
+    let response: Iresponse = {
+      status: 'success',
+      messages: ['Program record(s) successfully found'],
+      data: programs,
+    };
+    return response;
   }
 
-  findOne(id: number) {
-    return this.dbService.student.findFirst({
-      where: {
-        id: id,
-      },
-    });
+  async findOne(id: number) {
+    let data: Program = null;
+    const messages: string[] = [];
+    let status: string = '';
+
+    try {
+      data = await this.dbService.program.findUniqueOrThrow({
+        where: {
+          id: id,
+        },
+      });
+
+      messages.push("Requested student's record successfully found");
+      status = 'success';
+    } catch (error) {
+
+      messages.push("Requested program's record cannot be found");
+      status = 'failed';
+    } finally {
+
+      let response: Iresponse = {
+        status: status,
+        messages: messages,
+      };
```

Comment:
Please, replace the response structure according to my last push where I have used manageResponse() function